src/main/java/jcifs/smb1/UniAddress.java

```diff
@@ -0,0 +1,470 @@
+     *
+     * @param hostname NetBIOS or DNS hostname to resolve
+     * @throws java.net.UnknownHostException if there is an error resolving the name
+     */
+
+    public static UniAddress getByName( String hostname )
+                                        throws UnknownHostException {
+        return getByName( hostname, false );
+    }
+
+    static boolean isDotQuadIP( String hostname ) {
+        if( Character.isDigit( hostname.charAt( 0 ))) {
+            int i, len, dots;
+            char[] data;
+
+            i = dots = 0;                    /* quick IP address validation */
+            len = hostname.length();
+            data = hostname.toCharArray();
+            while( i < len && Character.isDigit( data[i++] )) {
+                if( i == len && dots == 3 ) {
+                    // probably an IP address
+                    return true;
+                }
+                if( i < len && data[i] == '.' ) {
+                    dots++;
+                    i++;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    static boolean isAllDigits( String hostname ) {
+        for (int i = 0; i < hostname.length(); i++) {
+            if (Character.isDigit( hostname.charAt( i )) == false) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Lookup <tt>hostname</tt> and return it's <tt>UniAddress</tt>. If the
+     * <tt>possibleNTDomainOrWorkgroup</tt> parameter is <tt>true</tt> an
+     * addtional name query will be performed to locate a master browser.
+     */
+
+    public static UniAddress getByName( String hostname,
+                                        boolean possibleNTDomainOrWorkgroup )
+                                        throws UnknownHostException {
+        UniAddress[] addrs = UniAddress.getAllByName(hostname, possibleNTDomainOrWorkgroup);
+        return addrs[0];
+    }
+    public static UniAddress[] getAllByName( String hostname,
+                                        boolean possibleNTDomainOrWorkgroup )
+                                        throws UnknownHostException {
+        Object addr;
+        int i;
+
+        if( hostname == null || hostname.length() == 0 ) {
+            throw new UnknownHostException();
+        }
+
+        if( isDotQuadIP( hostname )) {
+            UniAddress[] addrs = new UniAddress[1];
+            addrs[0] = new UniAddress( NbtAddress.getByName( hostname ));
+            return addrs;
+        }
+
+        for( i = 0; i < resolveOrder.length; i++ ) {
+            try {
+                switch( resolveOrder[i] ) {
+                    case RESOLVER_LMHOSTS:
+                        if(( addr = Lmhosts.getByName( hostname )) == null ) {
+                            continue;
+                        }
+                        break;
+                    case RESOLVER_WINS:
+                        if( hostname == NbtAddress.MASTER_BROWSER_NAME ||
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.